drivers/snmp-ups.c

```diff
@@ -3504,19 +3506,26 @@ bool_t su_ups_get(snmp_info_t *su_info_p)
 
 	if (!strcasecmp(su_info_p->info_type, "ups.status")) {
 /* FIXME: daisychain status support! */
-		upsdebugx(2, "%s: requesting nut_snmp_get_int() for "
-			"ups.status, with%s daisy template originally",
-			__func__, (format_char!=NULL ? "" : "out"));
-		status = nut_snmp_get_int(su_info_p->OID, &value);
-		if (status == TRUE)
+		if(!(strcasecmp(su_info_p->OID, ".1.3.6.1.4.1.318.1.1.1.11.1.1.0")))
 		{
-			su_status_set(su_info_p, value);
-			upsdebugx(2, "=> value: %ld", value);
+			status = nut_snmp_get_str(su_info_p->OID, buf, sizeof(buf), su_info_p->oid2info);
+			char pwr_status;
+			pwr_status = buf[1];
+			su_status_set(su_info_p, strtol(&pwr_status, NULL, 10));
+			if (status == TRUE)
+				upsdebugx(2, "=> value: %c", pwr_status);
+			else
+				upsdebugx(2, "=> Failed");
 		}
 		else
-			upsdebugx(2, "=> Failed");
-
-		free_info(tmp_info_p);
+		{
+			status = nut_snmp_get_int(su_info_p->OID, &value);
+			su_status_set(su_info_p, value);
+			if (status == TRUE)
+					upsdebugx(2, "=> value: %ld", value);
+			else
+					upsdebugx(2, "=> Failed");
+		}
```

Comment:
Why did this block change so drastically - a lot more than indenting generally, losing the debug log and now calling `su_status_set()` regardless of a successful `status` to fetch the value, for example? Did a local experiment seep out here?

Reply:
Alright I agree I did mess this up